src/odemis/driver/technolution.py

```diff
@@ -453,101 +519,64 @@ def __init__(self, name, role, parent, **kwargs):
-        setpoints_row = numpy.tile(numpy.concatenate((acq_delay_points, rise_points, flyback_points)), cell_size[1])
-        base = self.scanOffset.value[0] - 0.5 * cell_size[0] * scan_gain[0]
-        setpoints = base + numpy.concatenate((scan_delay_points, setpoints_row))
-
-        if setpoints.min() < -32768 or setpoints.max() > 32767:
-            raise ValueError("Setpoint values are to big/small to be handled by the ASM API")
-        return setpoints.astype(int).tolist()
-
-    def _getYAcqSetpoints(self):
-        scan_period = self.clockPeriod.value
-        scan_gain = self.scanGain.value
-        dwelltime = self.parent._ebeam_scanner.dwellTime.value  # s
-        scan_delay = self.parent._ebeam_scanner.scanDelay.value  # s
-        cell_size = self.parent._mppc.cellCompleteResolution.value  # pixels
-        acq_delay = self.parent._mppc.acqDelay.value - self.parent._ebeam_scanner.scanDelay.value[0]
+    def getXAcqSetpoints(self):
+        """
+        Creates the setpoints for the descanner in X direction (used by the ASM) for scanning one row of pixels. The
+        X_setpoints describe the movement of the descanner during the scanning of one full row of pixels.  To the ASM
+        API one period of setpoints (scanning of one row) is send which is repeated for all following rows.
+        A single sawtooth profile (rise and crash) followed by a flyback period (X=0) is used as trajectory. No
+        smoothing or low-pas filtering is used to create the trajectory of these setpoints.
 
-        # Function to find off beatness of the setpoint lines
-        # "(scan_period - input % scan_period) % scan_period" because when input is a multiple of scan_period
-        # the outcome should be 0, other wise the outcome could be the time left in that scan_period.
-        off_beat = lambda input: (scan_period - input % scan_period) % scan_period
+        :return: List of ints holding the setpoints in X direction.
+        """
+        descan_period = self.clockPeriod.value  # in seconds
+        dwelltime = self.parent._ebeam_scanner.dwellTime.value  # in seconds
+        X_scan_gain = self.scanGain.value[0]
+        X_scan_offset = self.scanOffset.value[0]
+        X_cell_size = self.parent._mppc.cellCompleteResolution.value[0]  # pixels
 
         # all units in seconds
-        #TODO K.K. if testcases in place change arange to zeros
-        scan_delay_points = 0 * numpy.arange(0, scan_delay[1], scan_period)
-        acq_delay_points = 0 * numpy.arange(off_beat(scan_delay[1]), acq_delay, scan_period)
-
-        rise_points = 0 * numpy.arange(off_beat(scan_delay[1] + acq_delay),
-                                                            dwelltime * cell_size[0], scan_period)
+        scanning_points = numpy.array(list(map(lambda t: (X_scan_gain / dwelltime) * t,
+                                               numpy.arange(0, dwelltime * X_cell_size, descan_period))))
```

Comment:
I'd recommend trying to write it using `numpy.linspace()`. One of the advantages is that you specify how many values you want (and so you're sure to not get one too many).

Reply:
I think that because we know here the step size an not the size of the array using linspace  the result would be more complicated. I think that part would become something like:
`numpy.linspace(0, dwelltime * X_cell_size, ((dwelltime * X_cell_size)//descan_period) + 1)`
While with arange the size is a result of the step size which is also how we would calculate the size.

Reply:
A little bit of explanation added to that, in the current setup we already now the step size (via the sampling frequencie/descan period) but not the number of sample points we will end up with.